source/distributed/shared_tria.cc

```diff
@@ -36,17 +36,39 @@ namespace parallel
     template <int dim, int spacedim>
     Triangulation<dim,spacedim>::Triangulation (MPI_Comm mpi_communicator,
                                                 const typename dealii::Triangulation<dim,spacedim>::MeshSmoothing smooth_grid,
-                                                const bool allow_artificial_cells):
+                                                const bool allow_artificial_cells,
+                                                const Settings settings):
       dealii::parallel::Triangulation<dim,spacedim>(mpi_communicator,smooth_grid,false),
+      settings (settings),
       allow_artificial_cells(allow_artificial_cells)
-    {}
+    {
+      Assert((settings & (partition_metis | partition_zorder | partition_custom_signal)) == partition_metis ||
+             (settings & (partition_metis | partition_zorder | partition_custom_signal)) == partition_zorder ||
+             (settings & (partition_metis | partition_zorder | partition_custom_signal)) == partition_custom_signal,
+             ExcMessage ("Settings must contain only one type of active cell partitioning scheme."))
+    }
 
 
 
     template <int dim, int spacedim>
     void Triangulation<dim,spacedim>::partition()
     {
-      dealii::GridTools::partition_triangulation (this->n_subdomains, *this);
+      if (settings & partition_metis)
+        {
+          dealii::GridTools::partition_triangulation (this->n_subdomains, *this);
+          if (settings & construct_multigrid_hierarchy)
```

Comment:
> we need an Assert for that, probably already there...

I don't understand. We don't want to call ``partition_multigrid_levels`` if the partitioning method is set to custom. This is implemented correctly in the current version of the PR. What do you want to assert?